src/gsy_e/models/strategy/heat_pump.py

```diff
@@ -0,0 +1,152 @@
+from gsy_e.models.strategy.trading_strategy_base import TradingStrategyBase
+from gsy_e.models.strategy.order_updater import OrderUpdaterParameters, OrderUpdater
+
+DEFAULT_HEAT_PUMP_ORDER_UPDATE_PARAMS = {
+    AvailableMarketTypes.SPOT: OrderUpdaterParameters(
+        initial_rate=ConstSettings.HeatPumpSettings.BUYING_RATE_RANGE.initial,
+        final_rate=ConstSettings.HeatPumpSettings.BUYING_RATE_RANGE.final,
+        update_interval=duration(
+                minutes=ConstSettings.GeneralSettings.DEFAULT_UPDATE_INTERVAL)),
+}
+
+if TYPE_CHECKING:
+    from gsy_e.models.market import MarketBase
+
+
+class HeatPumpStrategy(TradingStrategyBase):
+    """Strategy for heat pumps with storages."""
+    # pylint: disable=too-many-arguments)
+    def __init__(self,
+                 maximum_power_rating: float = ConstSettings.HeatPumpSettings.MAX_POWER_RATING_KW,
+                 min_temperatur_C: float = ConstSettings.HeatPumpSettings.MIN_TEMP_C,
+                 max_temperatur_C: float = ConstSettings.HeatPumpSettings.MAX_TEMP_C,
+                 external_temperatur_C: float = ConstSettings.HeatPumpSettings.EXT_TEMP_C,
+                 tank_volume_l: float = ConstSettings.HeatPumpSettings.TANK_VOL_L,
+                 average_consumption: float =
+                 ConstSettings.HeatPumpSettings.CONSUMPTION_KW,
+                 source_type: int = ConstSettings.HeatPumpSettings.SOURCE_TYPE,
+                 order_updater_parameters: Dict[
+                     AvailableMarketTypes, OrderUpdaterParameters] = None,
+                 preferred_buying_rate: float =
+                 ConstSettings.HeatPumpSettings.PREFERRED_BUYING_RATE
+                 ):
+        if not order_updater_parameters:
+            order_updater_parameters = DEFAULT_HEAT_PUMP_ORDER_UPDATE_PARAMS
+
+        super().__init__(order_updater_parameters=order_updater_parameters)
+
+        self._energy_params = HeatPumpEnergyParameters(
+            maximum_power_rating=maximum_power_rating,
+            min_temperatur_C=min_temperatur_C,
+            max_temperatur_C=max_temperatur_C,
+            external_temperatur_C=external_temperatur_C,
+            tank_volume_l=tank_volume_l,
+            average_consumption=average_consumption,
+            source_type=source_type
+        )
+
+        self.preferred_buying_rate = preferred_buying_rate
+
+    def serialize(self):
+        """Return serialised energy params."""
+        return {**self._energy_params.serialize()}
+
+    @property
+    def state(self) -> HeatPumpState:
+        return self._energy_params.state
+
+    def event_market_cycle(self) -> None:
+        super().event_market_cycle()
+        self._post_order_to_new_market(
+            self.area.spot_market, self.area.spot_market.time_slot)
+
+    def event_tick(self):
+        self._update_open_orders()
+
+    def event_bid_traded(self, *, _market_id: str, bid_trade: Trade) -> None:
+        """TODO: to be tested in the frame of GSYE-426"""
```

Comment:
I see, so you want to create a single test that will test the decrement of the posted energy functionality at the same time and not mock it, right? I agree then, this can be added in the context of the energy parameters ticket.

Reply:
Exactly!